src/NServiceBus.AcceptanceTesting/ScenarioContext.cs

```diff
@@ -28,10 +29,17 @@ public void AddTrace(string trace)
 
         ConcurrentQueue<string> traceQueue = new ConcurrentQueue<string>();
 
+        internal Dictionary<string, LogLevel> LogLevels = new Dictionary<string, LogLevel>();
+
+        public void SetLogLevel(string logger, LogLevel level)
```

Comment:
Sounds like a plan. I think I tried to optimize here because log gathering used to require cross-appdomain communication which is no longer the case. So you mean, append everything to the context and limit the trace output whatever is globally specified as log level (defaults to `info`)?